src/lib/services-manager/clients/services_manager.rb

```diff
@@ -0,0 +1,390 @@
+        end
+      end
+
+      # Table widget to show all services
+      #
+      # @return [Widgets::ServicesTable]
+      def services_table
+        @services_table ||= Widgets::ServicesTable.new(id: Id::SERVICES_TABLE)
+      end
+
+      # Buttons for actions over a selected service
+      #
+      # The log button only is included if YaST Journal is installed.
+      #
+      # @param service_name [String]
+      # @return [Yast::Term]
+      def service_buttons(service_name)
+        start_stop_label = ServicesManagerService.active(service_name) ? _('&Stop') : _('&Start')
+        start_mode_label = ServicesManagerService.start_mode_to_human_for(service_name)
+        buttons = [
+          PushButton(Id(Id::TOGGLE_RUNNING), start_stop_label),
+          HSpacing(1),
+          MenuButton(Id(Id::TOGGLE_ENABLED), start_mode_label, start_options_for(service_name)),
+          HStretch(),
+          PushButton(Id(Id::SHOW_DETAILS), _('Show &Details'))
+        ]
+
+        if journal_loaded?
+          buttons += [
+            HSpacing(1),
+            PushButton(Id(Id::SHOW_LOGS), _("Show &Log"))
+          ]
+        end
+
+        HBox(*buttons)
+      end
+
+      # Possible start mode options to select for a sevice
+      #
+      # @param service_name [String]
+      # @return [Array<Yast::Term>]
+      def start_options_for(service_name)
+        start_modes = ServicesManagerService.start_modes(service_name)
+
+        ServicesManagerService.all_start_modes.each_with_object([]) do |mode, all|
+          next unless start_modes.include?(mode)
+          all << Item(Id(mode), ServicesManagerService.start_mode_to_human(mode))
+        end
+      end
+
+      def handle_dialog
+        new_default_target = UI.QueryWidget(Id(Id::DEFAULT_TARGET), :Value)
+        Builtins.y2milestone("Setting new default target '#{new_default_target}'")
+        ServicesManagerTarget.default_target = new_default_target
+      end
+
+      def handle_table
+        if @prev_service != selected_service_name
+          @prev_service = selected_service_name
+          redraw_buttons(selected_service_name)
+        end
+      end
+
+      def save
+        Builtins.y2milestone('Writing configuration...')
+        UI.OpenDialog(Label(_('Writing configuration...')))
+        success = Yast::ServicesManager.save
+        UI.CloseDialog
+        if !success
+          success = ! Popup::ContinueCancel(
+            _("Writing the configuration failed:\n" +
+            Yast::ServicesManager.errors.join("\n") +
+            "\nWould you like to continue editing?")
+          )
+          Yast::ServicesManager.reset
+        end
+        success
+      end
+
+      # Reads services and updates the table content
+      #
+      # It shows a temporary popup meanwhile the services are obtained.
+      def redraw_services
+        UI.OpenDialog(Label(_('Reading services status...')))
+        services = services_names
+        UI.CloseDialog
```

Comment:
Yast2::Feedback is your friend - https://www.rubydoc.info/github/yast/yast-yast2/Yast2/Feedback#show-class_method